plugin/fzf.vim

```diff
@@ -75,7 +75,7 @@ function! s:shellesc_cmd(arg)
 endfunction
 
 function! fzf#shellescape(arg, ...)
-  let shell = get(a:000, 0, &shell)
+  let shell = get(a:000, 0, s:is_win ? 'cmd.exe' : &shell)
```

Comment:
Is this for `fzf#wrap` only? `fzf#run` calls `s:use_sh` to set the shell.

Reply:
It is for external program such as [junegunn/fzf.vim's vim script](https://github.com/junegunn/fzf.vim/blob/master/autoload/fzf/vim.vim) which seemingly calls this function before `fzf#run` execution.

Reply:
If you're doing this, go all the way. Change all invocations of `fzf#shellescape` within the context of `fzf#run` to pass `&shell`. Then

```
let shell = get(a:000, 0, s:is_win ? 'cmd.exe' : 'sh')
```

Reply:
I'm sorry but I can't get what exactly means to add `&shell` for all invocations of `fzf#shellescape`.
Could you tell me what aim is laid here, or what problems exist if I don't do that.

In my opinion, since `s:use_sh` have already replaces `&shell` to `sh` or `cmd.exe`,
there are not so much differences regardless of whether I do that or not.

Reply:
Idea is to not set the shell in `s:use_sh` on Windows in the future, similar to what vim-plug does, because there are many shell option combinations to consider depending on the OS and editor. vim-plug passes `&shell` to `fzf#shellescape` when it needs to run a shell command with the user's `&shell`.

Perhaps, it's too much to do it all in this PR. fzf supported cmd.exe on Windows so doing it halfway (ie. hardcode the default fallback shell to cmd.exe for `fzf#shellescape`) is fine for now.

Reply:
I partially introduced your suggestion at ca770ec63940fa665a283afc30c92bd3c9ffd9a4 for now.